frontend/src/components/Page.test.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+  This file is part of Edgehog.
+
+  Copyright 2023 SECO Mind Srl
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  SPDX-License-Identifier: Apache-2.0
+*/
+
+import { describe, it, expect, vi } from "vitest";
+import { screen } from "@testing-library/react";
+import { userEvent } from "@testing-library/user-event";
+import { renderWithProviders } from "setupTests";
+
+import Page from "./Page";
+
+describe("Header", () => {
+  it("renders title correctly", () => {
+    renderWithProviders(<Page.Header title="Header Title" />);
+
+    const pageTitle = screen.getByRole("heading", { name: "Header Title" });
+    expect(pageTitle).toBeVisible();
+    expect(pageTitle).toHaveTextContent("Header Title");
+  });
+
+  it("does not render title if title not given", () => {
+    renderWithProviders(<Page.Header />);
+
+    const pageTitle = screen.queryByRole("heading", { name: "" });
```

Comment:
Passing empty string in the `name` will ignore heading tags having text.
```suggestion
    const pageTitle = screen.queryByRole("heading");
```